src/datasets.c

```diff
@@ -719,8 +959,10 @@ int DatasetsInit(void)
                 Dataset *dset = DatasetGet(set_name, DATASET_TYPE_SHA256, save, load,
                         memcap > 0 ? memcap : default_memcap,
                         hashsize > 0 ? hashsize : default_hashsize);
-                if (dset == NULL)
-                    FatalError(SC_ERR_FATAL, "failed to setup dataset for %s", set_name);
+                if (dset == NULL) {
+                    FatalErrorOnInit(SC_ERR_FATAL, "failed to setup dataset for %s", set_name);
```

Comment:
this pattern will be confusing after init, as it will yell `FATAL` at the user, while acting as a warning.